src/java.desktop/share/classes/sun/java2d/cmm/lcms/LCMSTransform.java

```diff
@@ -149,10 +149,26 @@ private synchronized void doTransform(LCMSImageLayout in,
         LCMS.colorConvert(this, in, out);
     }
 
+    /**
+     * Returns {@code true} if lcms may supports this format directly.
+     */
+    private static boolean isLCMSSupport(BufferedImage src, BufferedImage dst) {
+        if (!dst.getColorModel().hasAlpha()) {
+            return true;
+        }
+        // lcms as of now does not support pre-alpha
+        if (src.isAlphaPremultiplied() || dst.isAlphaPremultiplied()) {
+            return false;
+        }
+        // lcms does not set correct alpha for transparent dst if src is opaque
+        // is it feature or bug?
+        return dst.getColorModel().hasAlpha() == src.getColorModel().hasAlpha();
+    }
+
```

Comment:
Thoughts about the method above:
 - I have left the first check as-is, so for any opaque destinations we will pass data to the lcms directly, but I am not sure that this is correct for source/destination with premultiplied alpha.
 - Since I am not sure about premultiplied alpha I have excluded it for the current fix if the destination has an alpha channel.